### ISSUE TITLE
Debugging VPN freezes with high load

### ISSUE THREAD

**Issue by maroneze** [adrienverge/openfortivpn#836]
I'm having some issues with my VPN provider since December, possibly related to the [Fortinet security update](https://www.fortinet.com/blog/psirt-blogs/update-regarding-cve-2018-13379); I have no idea if it is actually related to it, but my provider updated their server at this time, and since then I'm been having issues, while I had been using openfortivpn for years without such problems.

Namely, it seems that a high load (about 5MB/s, the limit of my wifi connection) causes a "freeze" in the connection, but without disconnecting; it just stops receiving packets, and I have to manually terminate the process and reconnect to get it working again. I'm using `openfortivpn` in the command-line with no special parameters. My `config` file only has `host`, `port`, `username` and `trusted-cert`.

It's not deterministic, so I've been trying to find which usage patterns trigger it. I tried adding `-vvv` to the command-line to see if I could spot any patterns, but no success so far. It seems, for instance, that Windows Update triggers it more often than other processes, possibly due to maximal load, but I see nothing in the terminal, other than a lot of small packets e.g. `pppd ---> gateway (62 bytes)`, due to my machine trying to contact the VPN server, but without getting any responses.

I wonder if there's something else I could do to debug it; for instance, to understand why the connection "stops" without disconnecting. It's possibly not even related to openfortivpn itself; in that case, is there something I could do to ascertain that?

**Comment by DimitriPapadopoulos** [comment]
Chances are we are connecting to the same provider :-) If so, I doubt this issue is specific to the VPN server, as I haven't been experiencing such issues even under high loads (including Windows Update running in a virtual machine).

Is it possible this issue appeared after an _openfortivpn_ upgrade? Which Linux distribution and _openfortivpn_ version is this?

**Comment by maroneze** [comment]
I'm currently using commit `v1.15.0`. I was using an older one (v1.14, I think), and upgraded when the issues started, in the hope that it would help fixing them. I was using the Fedora 32 package, but now I compile openfortivpn from the Github repository.

I tried using the "official" forticlient Debian binary, and it seemed more stable, but the same behavior happened with it once or twice. So, it's not exclusive to openfortivpn, but it does seem to happen more often (again, this might be simply because I use it more often than the other).

Overall, if there are some simple techniques to debug and try to pinpoint the cause, I'm interested in trying them; otherwise, since I have no evidence the cause is related to openfortivpn, I'll close the issue.

**Comment by DimitriPapadopoulos** [comment]
Since you are able to build from GitHub, I would recommend comparing _openfortivpn_ 1.14 (or the older version you used to run before December) and 1.15. For example run `date ; openfortivpn ...` and keep tabs on the "time to freeze" of _openfortivpn_, for both versions. If you do see a difference, then we can investigate and maybe identify regressions.

**Comment by DimitriPapadopoulos** [comment]
Can you confirm we connect to the same provider (celui de l'organisme auquel appartient le LIST)? If so, because the connection is quite stable for me, this might be an issue with this specific Linux machine, perhaps a driver issuer. I would also recommend trying the same version of _openfortivpn_ with a different computer if possible at all (one with different components and hence different drivers).

**Comment by maroneze** [comment]
Thanks for the information, indeed it must be unrelated to openfortivpn. I did try some older versions but didn't see much difference. I'll close it for now, thanks. If I ever see something specific or have better information I'll try debugging it.

**Comment by maroneze** [comment]
For the record, it seems it was an IPv6 issue. I had disabled it in the past, but some update must have re-enabled it at around the same time that the Fortigate server was upgraded (or the upgrade did change the behavior w.r.t. IPv6). Then, many times when DNS requests were attempted (mostly under high load, but not only), something would happen that would prevent the VPN from working at all.

I disabled once more IPv6 globally on my Fedora, and so far it seems to be stable; no more lost connections for a few days already.
